tests/test_when.py

```diff
@@ -485,6 +485,7 @@ class _Test(m.Circuit):
     assert check_gold(__file__, f"{basename}.mlir")
 
 
+@pytest.mark.skip("There's a bug in this test")
```

Comment:
Turns out there's a bug in this test, `x` is defined inside a when, and then referred to in the following otherwise, I'm not sure what the original intention of this was, it might have been a temporary change while I was debugging.  Working on a fix but will put it in a separate PR for clarity.